external/fv3fit/fv3fit/reservoir/validation.py

```diff
@@ -0,0 +1,204 @@
+            hybrid_inputs_time_series = get_ordered_X(
+                batch_data, model.hybrid_variables  # type: ignore
+            )
+            hybrid_inputs_time_series = _get_states_without_overlap(
+                hybrid_inputs_time_series, overlap=model.rank_divider.overlap
+            )
+        else:
+            hybrid_inputs_time_series = None
+
+        batch_predictions, batch_imperfect_predictions = _get_predictions_over_batch(
+            model, states_with_overlap_time_series, hybrid_inputs_time_series
+        )
+
+        one_step_prediction_time_series += batch_predictions
+        one_step_imperfect_prediction_time_series += batch_imperfect_predictions
+        target_time_series.append(
+            _get_states_without_overlap(
+                states_with_overlap_time_series, overlap=model.rank_divider.overlap
+            )
+        )
+    target_time_series = np.concatenate(target_time_series, axis=0)[n_synchronize:]
+
+    persistence = target_time_series[:-1]
+    target = np.array(target_time_series[1:])
+
+    # _get_predictions_over_batch predicts up to n_timesteps-1
+    one_step_predictions = np.array(one_step_prediction_time_series)[n_synchronize:-1]
+    time_means_to_calculate = {
+        "time_mean_prediction": one_step_predictions,
+        "time_mean_prediction_error": one_step_predictions - target,
+        "time_mean_persistence_error": persistence - target,
+        "time_mean_prediction_mse": (one_step_predictions - target) ** 2,
+        "time_mean_persistence_mse": (persistence - target) ** 2,
+    }
+
+    if len(one_step_imperfect_prediction_time_series) > 0:
+        one_step_imperfect_predictions = np.array(
+            one_step_imperfect_prediction_time_series
+        )[n_synchronize:-1]
+        imperfect_prediction_time_means_to_calculate = {
+            "time_mean_imperfect_prediction": one_step_imperfect_predictions,
+            "time_mean_imperfect_prediction_error": one_step_imperfect_predictions
+            - target,
+            "time_mean_imperfect_prediction_mse": (
+                one_step_imperfect_predictions - target
+            )
+            ** 2,
+        }
+        time_means_to_calculate.update(imperfect_prediction_time_means_to_calculate)
+
+    diags_ = []
+    for key, data in time_means_to_calculate.items():
+        diags_.append(_time_mean_dataset(model.input_variables, data, key))
+
+    return xr.merge(diags_)
+
+
+UNITS = {
+    "air_temperature": "K",
+    "specific_humidity": "kg/kg",
+    "air_pressure": "Pa",
+    "eastward_wind": "m/s",
+    "northward_wind": "m/s",
+}
+
+
+def log_rmse_z_plots(ds_val, variables):
+    for var in variables:
+        rmse = {}
+        for comparison in ["persistence", "prediction", "imperfect_prediction"]:
+            mse_key = f"time_mean_{comparison}_mse_{var}"
+            if mse_key in ds_val:
+                rmse[comparison] = np.sqrt(ds_val[mse_key].mean(["x", "y"])).values
```

Comment:
Doesn't matter for comparison, but if we ever use these values for a publication we should make a note at least to switch to area-weighted spatial averages.